pyproject.toml

```diff
@@ -2,29 +2,116 @@
 name = "hermes"
 version = "0.1.0"
 description = "Workflow to publish research software with rich metadata"
-authors = ["Stephan Druskat <stephan.druskat@dlr.de>", "Michael Meinel <michael.meinel@dlr.de>"]
+homepage = "https://software-metadata.pub"
+license = "Apache-2.0"
+authors = [
+    "Stephan Druskat <stephan.druskat@dlr.de>",
+    "Michael Meinel <michael.meinel@dlr.de>",
+    "Oliver Bertuch <o.bertuch@fz-juelich.de>",
+    "Jeffrey Kelling <j.kelling@hzdr.de>",
+    "Oliver Knodel <o.knodel@hzd.r.de>"
+]
 
-[tool.poetry.extras]
-cff = ['ruamel.yaml', 'cffconvert']
+readme = "README.md"
+repository = "https://github.com/hermes-hmc/workflow"
+documentation = "https://docs.software-metadata.pub"
+keywords = ["publishing", "metadata", "automation"]
+
+include = [
+    "hermes/schema/*.json",
+]
 
 [tool.poetry.dependencies]
 python = "^3.10"
 click = "^8.1"
 "ruamel.yaml" = "^0.17.21"
 jsonschema = "^3.0.0"
 cffconvert = "^2.0.0"
+toml = "^0.10.2"
+Markdown = "^3.4.1"
 
 [tool.poetry.dev-dependencies]
 pytest = "^7.1.1"
 pytest-cov = "^3.0.0"
 Sphinx = "^4.5.0"
 myst-parser = "^0.17.2"
+sphinx-book-theme = "^0.3.3"
+sphinx-favicon = "^0.2"
+sphinxcontrib-contentui = "^0.2.5"
+sphinxcontrib-images = "^0.9.4"
+sphinx-icon = "^0.1.2"
+sphinxemoji = "^0.2.0"
+sphinxext-opengraph = "^0.6.3"
 
 [tool.poetry.plugins.console_scripts]
 haggis = "hermes.cli:haggis"
 
 [tool.poetry.plugins."hermes.harvest"]
-cff = "hermes.commands.harvest.cff:harvest_cff"
+000_cff = "hermes.commands.harvest.cff:harvest_cff"
+010_codemeta = "hermes.commands.harvest.codemeta:harvest_codemeta"
+020_git = "hermes.commands.harvest.git:harvest_git"
```

Comment:
Maybe this i a point to bring this up: Why do we need this aliasing here? Why do we not just structure the module(s) in such a way that we do not the aliases, like plain old python modules?

I stumbled over this back when I added the git harverster. It confuses me, because this appears to needlessly add a requirement to manually replicate information in two places (here and in the filesystem/source tree) .. which seems especially weird as we are working on a convenience tool for meta-data handling here. I am not familiar with poetry ... so, is there a good reason for this?

Reply:
Well, these are not "aliases" but so-called entry points.

Python provides a method for extension of of your own code base - you can think of it like good old "plug-ins". This gives the opportunity of using an automated process of loading these Python functions without explicitly handling all the caveats that come with it. You can especially use these entry points to provide additional features from thrid-party Python packages. This would not be possible without entry points. On top, these entry points are a well-defined standard within Python.

The names here are in fact not very nice ... this will change as soon as we have a working configuration system. However, according to the specification of entry points, they need a name.

I hope that clearifies why this is needed.